### PR TITLE
feat(api): Build alert rule details api (SEN-872)

### PR DESCRIPTION
**Pull request by wedamija**
Implements an api to fetch details for an individual alert rule.

### CODE REVIEW

File: src/sentry/incidents/endpoints/bases.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        try:
+            kwargs['alert_rule'] = AlertRule.objects.get(
+                project=project,
+                id=alert_rule_id,
+            )
+        except AlertRule.DoesNotExist:
+            raise ResourceDoesNotExist
+
+        if not request.access.has_project_access(project):
+            raise PermissionDenied
```
Comment (markstory) [inline src/sentry/incidents/endpoints/bases.py]: Should this check be done before fetching the alert rule? When the user doesn't have access you will save a SQL query and not disclose an alert rule's existence.

File: tests/sentry/incidents/endpoints/test_project_alert_rule_details.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+from sentry.api.serializers import serialize
+from sentry.incidents.logic import create_alert_rule
+from sentry.incidents.models import (
+    AlertRuleAggregations,
+    AlertRuleThresholdType,
+)
+from sentry.testutils import APITestCase
+
+
+class AlertRuleListEndpointTest(APITestCase):
```
Comment (markstory) [inline tests/sentry/incidents/endpoints/test_project_alert_rule_details.py]: Should this be AlertRuleDetailsEndpointTest?

Comment (wedamija) [reply tests/sentry/incidents/endpoints/test_project_alert_rule_details.py]: 😬

**Review by markstory — APPROVED** [review]
Looks good to me.

**Pull request merged**